### PR TITLE
Plugin/etcd: Only write config values when not already correct

### PR DESCRIPTION
**Pull request by neiljerram**

### CODE REVIEW

**Comment by neiljerram** [comment]
Addresses https://github.com/Metaswitch/calico/issues/374

**Comment by ed-harrison** [comment]
LGTM

File: calico/openstack/t_etcd.py [hunk 1]
```diff
@@ -352,11 +352,25 @@ def security_group_updated(self, sg):
...
+        """
+        # First read the config values, so as to avoid unnecessary
+        # writes.
+        prefix = None
+        ready = None
+        try:
+            prefix = self.client.read('/calico/config/InterfacePrefix').value
+            ready = self.client.read('/calico/config/Ready').value
+        except Exception:
+            pass
```
Comment (Lukasa) [inline calico/openstack/t_etcd.py]: At the very least you want to log here, so that we can work out that exceptions are occurring if we hit them live.

**Comment by Lukasa** [comment]
One note from me.

**Comment by neiljerram** [comment]
Delta pushed.

**Pull request merged**